main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    each do |i|
+      yield i
+    end
+    self
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    0.upto(length - 1) do |i|
+      yield i, index
+      index + 1
+    end
+  end
+
+  def my_select
+    return unless block_given?
+
+    result = []
+    my_each do |i|
+      result.push(i) if yield i
+    end
+    result
+  end
+
+  def my_all?(*val)
+    result = true
+    if !val[0].nil?
+      my_each { |i| result = false unless val[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false unless yield(i) }
+    end
+    result
+  end
+
+  def my_any?(*val)
+    result = true
+    if !val[0].nil?
+      my_each { |i| result = false unless val[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false unless yield(i) }
+    end
+    result
+  end
```

Comment:
# my_any?
- [ ] `my_any?` returns false if the block ever returns a value false or nil::range

      Failure/Error: expect(range.my_any?(&false_block)).to eq(range.any?(&false_block))

        expected: true
             got: false

##
- [ ] `my_any?` when no block or argument is given returns true if at least one of the collection is not false or nil

      Failure/Error: expect(true_array.my_any?).to be true_array.any?

        expected true
             got false
 ##
- [ ] `my_any?` when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric

      Failure/Error: expect(array.my_any?(Numeric)).to be array.any?(Numeric)

        expected true
             got false
##      
- [ ] `my_any?` when a Regex is passed as an argument returns true if any of the collection matches the Regex

      Failure/Error: expect(words.my_any?(/d/)).to be words.any?(/d/)

        expected true
             got false
##      
- [ ] `my_any?` when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern

      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')

        expected true
             got false